### ISSUE TITLE
False negative: invalid generic in function argument is accepted

### ISSUE THREAD

**Issue by c1887** [astral-sh/ty#3176]
### Summary

Good morning,

I asked this on the Discord server before and was asked to open an issue:

[Playground link](https://play.ty.dev/41eb7b97-532e-450b-a6b1-5d867f9eac08)

Minimal working example:
```python
class A: ...


class B: ...


class F[T: A | B]: ...


def foo(f: F[A]) -> F[A]:
    # do something with f
    return f


f: F[B] = F[B]()
foo(f)  # error in mypy, is fine in ty
```

As written in the comment, I'd expect this to raise an invalid argument error.

### Version

ty 0.0.17 (8cec85718 2026-02-13)

**Comment by sharkdp** [comment]
Thank you for reporting this.

This passes because `F` is currently inferred as being bivariant in `T`. See https://github.com/astral-sh/ty/issues/1728 for details.
